add basic Apache Parquet implementation

martenson opened the pull request:
ping @xtracko  - do we want anything more for a start?

guerler committed:
Hide scratchbook when last window has been closed

guerler committed:
Fix note class in masthead

guerler committed:
Remove conflicting style definition, fix alignment of notes

guerler committed:
Revise active tag handling, fix for dropdowns

guerler committed:
Remove parsing of galaxy instance

guerler committed:
Use local instead of parsed property in item

guerler committed:
Remove galaxy and root parsing in masthead mounting helper

guerler committed:
Disable shared data if user is not logged in, regression

guerler committed:
Fix test cases, ensure active class to be set properly

guerler committed:
Fix href masthead test cases

guerler committed:
Restore anonuser access to shared data

guerler committed:
Run prettier

guerler committed:
Remove unused masthead backbone wrapper vue component

kjetilkl committed:
Added handler for history options menu webhook

jdavcs committed:
Add custom activation email message

jmchilton committed:
Remove migration logic that will never be used.

- Remove all logic from 0005_cleanup_datasets_fix.py. Noone is upgrading from a Galaxy running db version 0004 to 21.01.
- Remove transfer job stuff - I would be shocked if that ever did something useful on anything but main/test. At any rate it certainly won't do anything useful at this point.

bgruening committed:
Merge branch 'dev' into cleanup_migration

nsoranzo committed:
Move framework tests to GitHub workflows

martenson committed:
add basic Apache Parquet implementation

xtracko commented:
I would say, that this is fairly sufficient 👍 

Albeit, we have the Query tool to handle the format conversions, it would be nice to add a native converter to/from CSV. Unfortunately, I can't imagine the required amount of work. In case, the Galaxy is already dependant on the Pandas package, this could be solved pretty easily. But otherwise, doing it without Pandas could be tedious work.

mvdbeek committed:
Merge pull request #10802 from guerler/close_empty_scratchbook

Close empty multi view, fix icon alignment

jmchilton committed:
Merge pull request #10949 from kjetilkl/history_options_webhook

Added handler for history options menu webhook

jmchilton committed:
Merge pull request #10959 from ic4f/dev_issue10607

Add custom activation email message

jmchilton committed:
Optimized binary datatype sniffing.

jmchilton committed:
Merge pull request #11038 from jmchilton/cleanup_migration

Remove migration logic that will never be used now.

dependabot[bot] committed:
Bump axios from 0.19.2 to 0.21.1 in /client

Bumps [axios](https://github.com/axios/axios) from 0.19.2 to 0.21.1.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v0.21.1/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v0.19.2...v0.21.1)

Signed-off-by: dependabot[bot] <support@github.com>

dannon committed:
Merge pull request #11053 from galaxyproject/dependabot/npm_and_yarn/client/axios-0.21.1

Bump axios from 0.19.2 to 0.21.1 in /client

nsoranzo committed:
Lint a bunch of test tools

nsoranzo committed:
Fix API tests failing when usegalaxy.org is down

`site_down` was broken when a `requests.get()` raises a timeout exception
(should have returned `True`), so I've inverted the logic to check if
the site is up, which is more natural.

Added a proper test for uploading an invalid URL, renamed the one which was
actually testing a 404 HTTP error.

jmchilton committed:
Apply suggestions from code review

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

nsoranzo committed:
Merge pull request #11049 from jmchilton/binary_sniffing_prefixes

jmchilton committed:
Merge pull request #11055 from nsoranzo/framework_CI

Move framework tests to GitHub workflows + lint test tools

martenson committed:
reflect recent changes in magic number sniffing

martenson commented:
refactored and merged upstream in https://github.com/galaxyproject/galaxy/pull/11047, we should get this with 21.01
